src/main/java/practice/kadai2209_7th/AirportController.java

```diff
@@ -18,115 +19,161 @@
 @RestController
 @RequestMapping("/airports")
 public class AirportController {
+
+    List<String> airportCodeList = new ArrayList<>();
+
     Map<String, List<String>> yourAirportMap = new HashMap<>();
 
-    private boolean hasAirportList = false;
+    private final String notFoundMessage = "not found";
+
     @Autowired
     private PortNumber portNum;
 
     @Autowired
     private AirportService service;
 
-    @GetMapping("/airports")
-    public String getAirportList() {
+    AirportAllData airportAllData = new AirportAllData();
 
-        if (!hasAirportList) {
 
-            yourAirportMap = service.getAllDataMap();
+    @ResponseStatus(code = HttpStatus.OK)
+    @GetMapping("/airports")
+    public Map<String, Object> getAirportList() {
 
-            hasAirportList = true;
-        }
+        airportCodeList = service.getAllAirportCode();
 
-        return "You have the list of airports";
+        Map<String, Object> airportCodeMap = new LinkedHashMap<>();
+        airportCodeMap.put("message", "You have airport codes listed below");
+        airportCodeMap.put("airport", airportCodeList);
+
+        return airportCodeMap;
     }
 
+
     @GetMapping("/search")
-    public ResponseEntity<List<String>> getAirports(
-            @RequestParam(value = "airportCode"/*, defaultValue = "KIX"*/) @Size(min = 3, max = 3, message = "Number of letters has to be 3")
-            String airportCode) {
+    public ResponseEntity<Map<String, Object>> getAirportMap(
+            @RequestParam(value = "airportCode", defaultValue = "NRT") @Size(min = 3, max = 3, message = "Number of letters has to be 3") String airportCode) {
 
-        if (yourAirportMap.isEmpty()) {
+        yourAirportMap = airportAllData.getAirportAllData(yourAirportMap, service);
```

Comment:
これもフィールド変数にしないで都度変数化すればよいです。
```suggestion
       Map<String, List<String>> yourAirportMap = airportAllData.getAirportAllData(yourAirportMap, service);
```

Reply:
ご指摘の個所を修正致しました。
（変数のスコープをメソッド内に限定、ただし`yourAirportMap`については、このままだとPOST/PATCH/DELETEによるMapの情報を保持できないため、別途補完）
a9e2f69921863a807122e4a3bffe51ab370bf5d1

![image](https://user-images.githubusercontent.com/113277395/198876311-5e9dd229-2ac4-4e89-abe3-89e5b28b1be4.png)
--
![image](https://user-images.githubusercontent.com/113277395/198876082-1085c51f-9589-4e7b-a3a1-16786d840340.png)

Reply:
`yourAirportMap`のエンティティ追加/更新/削除結果を保持するため



----
### 上記に伴い、Controller内のメソッドを修正
![image](https://user-images.githubusercontent.com/113277395/199878609-67db58e7-d46a-48d0-883e-7f2fad4b518f.png)
![image](https://user-images.githubusercontent.com/113277395/199878642-4ea38611-dc58-4b16-a1dd-ebf1264222f2.png)
--
### エンティティ追加/更新メソッド
![image](https://user-images.githubusercontent.com/113277395/199878736-89dd6162-8358-4869-ad2b-cecf9e382c78.png)
--
### エンティティ追加メソッド
![image](https://user-images.githubusercontent.com/113277395/199878819-926f6a13-9de8-4abe-ba94-c840c92e7f37.png)
c3029c3a29e72493794f80b93c1d0062df908d1d